### ISSUE TITLE
Add MultiThread TreeWalker implementation

### ISSUE THREAD

**Issue by soon** [checkstyle/checkstyle#4957]
Since walking is one of the most consuming operations, we should process checks in parallel inside the MT version of TreeWalker

**Comment by jo2** [comment]
Can someone give a status update here? I'm running into multiple problems with synchronization in version 8.26. The GitHub project mentioned here is closed but the issue is stil open and in ToDO..

If there is no work in progress I work on some PRs fixing the problems I'll run into..

**Comment by romani** [comment]
Checkstyle is single thread tool now, we did not come on agreement on MT mode, so postponed such discussion till we have time to look at it one more time.

**Comment by rnveach** [comment]
@jo2 since there is no multi-threading right now, I recommend creating a new issue for your problem.
